src/lib/config-utils.ts

```diff
@@ -0,0 +1,376 @@
+		}
+		return val;
+	});
+
+	return parsedEnv as ConfigOptions;
+}
+
+export function bootConfigToEnv(deviceType: string, config: ConfigOptions): EnvVarObject {
+	const prefix = bootConfigVarPrefix(deviceType);
+	const confWithEnvKeys = _.mapKeys(config, (_val, key) => {
+		return prefix + key;
+	});
+	return _.mapValues(confWithEnvKeys, (val) => {
+		if (_.isArray(val)) {
+			return JSON.stringify(val).replace(/^\[(.*)\]$/, '$1');
+		}
+		return val;
+	});
+}
+
+export function filterConfigKeys(
+	deviceType: string,
+	allowedKeys: string[],
+	conf: { [key: string]: any },
+): { [key: string]: any } {
+
+	let isConfigType: boolean = false;
+	let prefix: string;
+	if (isConfigDeviceType(deviceType)) {
+		prefix = bootConfigVarPrefix(deviceType);
+		isConfigType = true;
+	}
+	return _.pickBy(conf, (_v, k) => {
+		return _.includes(allowedKeys, k) || (isConfigType && _.startsWith(k, prefix));
+	});
+}
+
+export function getBootConfigPath(deviceType: string): string {
+	if (isRPiDeviceType(deviceType)) {
+		return getRpiBootConfig();
+	} else if (isExtlinuxDeviceType(deviceType)) {
+		return getExtlinuxBootConfig();
+	} else {
+		throw new Error(`No boot config exists for device type: ${deviceType}`);
+	}
+}
+
+function getRpiBootConfig(): string {
+	return bootMountPoint + '/config.txt';
+}
+
+function getExtlinuxBootConfig(): string {
+	return bootMountPoint + '/extlinux/extlinux.conf';
+}
+
+export function parseBootConfig(deviceType: string, conf: string): ConfigOptions {
+	if (isRPiDeviceType(deviceType)) {
+		return parseRpiBootConfig(conf);
+	} else if (isExtlinuxDeviceType(deviceType)) {
+		return parseExtlinuxBootConfig(conf);
+	} else {
+		throw new Error(`Cannot parse boot config for device type: ${deviceType}`);
+	}
+}
+
+export function parseRpiBootConfig(confStr: string): ConfigOptions {
+	const conf: ConfigOptions = { };
+	const configStatements = confStr.split(/\r?\n/);
+
+	for (const configStr of configStatements) {
+		// Don't show warnings for comments and empty lines
+		const trimmed = _.trimStart(configStr);
+		if (_.startsWith(trimmed, '#') || trimmed === '') {
+			continue;
+		}
+		let keyValue = /^([^#=]+)=(.+)/.exec(configStr);
+		if (keyValue != null) {
+			if (!_.includes(rpiArrayConfigKeys, keyValue[1])) {
+				conf[keyValue[1]] = keyValue[2];
+			} else {
+				const key = keyValue[1];
+				if (conf[key] == null) {
+					conf[key] = [];
+				}
+				(conf[key] as string[]).push(keyValue[2]);
+			}
+		} else {
+			keyValue = /^(initramfs) (.+)/.exec(configStr);
+			if (keyValue != null) {
+				conf[keyValue[1]] = keyValue[2];
+			} else {
+				console.log(`Warning - Could not parse config.txt entry: ${configStr}. Ignoring.`);
+			}
+		}
+	}
+
+	return conf;
+}
+
+export function parseExtlinuxBootConfig(confStr: string): ConfigOptions {
+	const parsedBootFile = parseExtlinuxFile(confStr);
+
+	// First find the default label name
+	const defaultLabel = _.find(parsedBootFile.globals, (_v, l) => {
+		if (l === 'DEFAULT') {
```

Comment:
Maybe compare lower-case (or upper-case)? And you can actually write this as:

```
const defaultLabel = _.find(parsedBootFile.globals, (_v, l) => l.toUpperCase() === 'DEFAULT');
```

Reply:
I'm not worried about this too mcuh due to https://github.com/resin-io/resin-supervisor/pull/672/files/89627a33925e218275196bd5fe7b193e5e2737c7#diff-420116c700357465a04fcc85c5baa00eR258